### PR TITLE
remove self.init_args

### PR DESCRIPTION
**Pull request by CamDavidsonPilon**
Correct me if I am wrong, but I don't think `self.init_args` is needed everywhere. `copy.copy` should transfer the properties over. This improves the maintainability of the code, and makes adding strategies easier. 

For review @meatballs et al.

### CODE REVIEW

File: axelrod/player.py [hunk 1]
```diff
@@ -119,9 +118,8 @@ def clone(self):
         """Clones the player without history, reapplying configuration
         parameters as necessary."""
 
-        cls = self.__class__
-        new_player = cls(*self.init_args)
-        new_player.tournament_attributes = copy.copy(self.tournament_attributes)
+        new_player = copy.copy(self)
```
Comment (CamDavidsonPilon) [inline axelrod/player.py]: this may need to be `copy.deepcopy`

**Comment by CamDavidsonPilon** [comment]
I imagine a few tests will fail, I'm just looking for feedback before I proceed.

**Comment by marcharper** [comment]
Hi @CamDavidsonPilon ! Thanks for working on the library.

The presence of `init_args` is intentional. We have tried using copy before and there are some issues to consider:
* For some strategies one likely needs to use `copy.deepcopy` (like the meta strategies), since the internal variables can contain objects (like the `team` variable of a MetaPlayer)
* `deepcopy` can be slow
* There may be hard to track bugs due to copying that the tests do not cover (like retaining some kind of state between rounds)

That said, if you can make it work without a big performance hit, we'll certainly consider merging!

**Comment by CamDavidsonPilon** [comment]
Hmm, ok, from there:

1. could the more complicated strategies reimplement their own `clone` function?
2. Agree - I bet `copy.copy` could be used for most cases that didn't require reimplementing `clone`.
3. Yes, I wonder what we would uncover. `copy.copy` should avoid these though.

I'll keep thinking about this.

**Comment by marcharper** [comment]
It's a tricky part of the code. We are also having issues with the deepcopy in MindReader -- it's really slow when paired with some specific strategies. If you have any thoughts on that please do share :)

**Comment by drvinceknight** [comment]
> The presence of init_args is intentional. We have tried using copy before and there are some issues to consider:

Is it worth having this in a docstring somewhere perhaps?

**Comment by marcharper** [comment]
Yes, or a comment in the code I think, or in the docs...

**Comment by CamDavidsonPilon** [comment]
I'll close this for now

**Pull request closed without merging**